rpc/src/rpc_url.rs

```diff
@@ -98,15 +99,22 @@ impl Url {
     }
 
     /// Get the username associated with this URL, if any.
-    pub fn username(&self) -> &str {
-        self.inner.username()
+    pub fn username(&self) -> Option<&str> {
+        Some(self.inner.username()).filter(|s| !s.is_empty())
     }
 
     /// Get the password associated with this URL, if any.
     pub fn password(&self) -> Option<&str> {
         self.inner.password()
     }
 
+    /// Get the authority associated with this URL, if any.
+    /// The authority is the username and password separated by a colon.
+    pub fn authority(&self) -> Option<String> {
+        self.username()
+            .map(|user| format!("{user}:{}", self.password().unwrap_or_default()))
```

Comment:
Seems like Rust nightly doesn't like this?

Reply:
Fixed